### PR TITLE
Refactor

### LINKED ISSUE #85: Refactor all code
- Use `std::unique_lock` instead of lock and unlock calls.
- Rename `speed_scale` variable on line 187 of publish_control.cpp. Also look for other variable re-naming.
- Remove any excessive/unhelpful comments.
- Remove need to specify static member variables in `publish_control.cpp`.
- auto-format all code.

### PR DESCRIPTION
**Pull request by icolwell-as**
Resolves #85 by changing around some things and auto-formatting all code.
Generally functionality should remain unchanged.

### CODE REVIEW

File: include/pacmod_game_control/globals.h [hunk 1]
```diff
@@ -17,10 +17,6 @@
 #include <std_msgs/Bool.h>
 #include <pacmod_msgs/VehicleSpeedRpt.h>
 
-namespace AS
-{
-namespace Joystick
-{
 enum GamepadType
```
Comment (jilinzhouas) [inline include/pacmod_game_control/globals.h:20]: enum GamepadType : uint8_t {
...
}

Comment (jilinzhouas) [inline include/pacmod_game_control/globals.h:20]: use enum class GamepadType or enum GamepadType : uint8_t and give value to each entry.

File: include/pacmod_game_control/publish_control.h [hunk 2]
```diff
@@ -19,93 +19,99 @@
 #include <pacmod_msgs/SystemCmdInt.h>
 #include <pacmod_msgs/SystemRptInt.h>
 
-namespace AS
```
Comment (jilinzhouas) [inline include/pacmod_game_control/publish_control.h:22]: Any reason to remove the namespaces?

File: include/pacmod_game_control/publish_control.h [hunk 3]
```diff
@@ -19,93 +19,99 @@
...
-    static bool headlight_state_change;
-    static int wiper_state;
-    static int last_shift_cmd;
-    static int last_turn_cmd;
-    static int last_rear_pass_door_cmd;
-    static float last_brake_cmd;
+  // public variables
+  JoyAxis steering_axis = LEFT_STICK_LR;
+  float max_rot_rad = MAX_ROT_RAD_DEFAULT;
+  int vehicle_type = INVALID;
```
Comment (jilinzhouas) [inline include/pacmod_game_control/publish_control.h:37]: Should it be VehicleType?

File: include/pacmod_game_control/globals.h [hunk 4]
```diff
@@ -80,37 +76,26 @@ struct EnumHash
...
-static std::mutex speed_mutex;
-static std::mutex state_change_mutex;
-static std::mutex shift_mutex;
-static std::mutex turn_mutex;
-static std::mutex rear_pass_door_mutex;
-
-}  // namespace Joystick
-}  // namespace AS
+// constants
+const float ROT_RANGE_SCALER_LB = 0.05;
```
[the remark this answers is not in the document]
Comment (icolwell-as) [reply include/pacmod_game_control/globals.h:80]: I just didn't see any need/benefit to having the namespaces. All it did is result in more verbose typing or `using` statements. Mostly everything is already encapsulated inside the `PublishControl` class.
Am I missing some potential benefit to having it namespaced?

[the remark this answers is not in the document]
Comment (jilinzhouas) [reply include/pacmod_game_control/globals.h:80]: The main idea of namespace is to organize the code nicely and avoid name clashes. For this application, probably we do not see much benefits of the namespace usage. If we are providing some code which can be used/reused by third party, it is always good to have them inside a namespace. If I guess correctly, all of our public code in github has some namespaces (AS/xxx/xx) defined for them. If some variables/consts are just used by one source file, we can put them inside an unamed namespace so they only can be referenced from that source file.

[the remark this answers is not in the document]
Comment (icolwell-as) [reply include/pacmod_game_control/globals.h:80]: Thanks, ya, all the code in this repo is only ever used inside this repo. There's really no use case I can see that would require this code used in some other application/way.

File: src/startup_checks.cpp [hunk 5]
```diff
@@ -1,34 +1,49 @@
...
+{
+  bool willExit = false;
+
+  ros::NodeHandle pnh("~");
+
+  // Run startup checks
+  willExit = check_steering_stick_left_right(pnh);
+  willExit = check_vehicle_type(pnh);
+  willExit = check_controller_type(pnh);
+  willExit = check_scale_values(pnh);
```
Comment (shan-as) [inline src/startup_checks.cpp:25]: Wouldn't the last `willExit` overwrite the result of previous results?

Comment (icolwell-as) [reply src/startup_checks.cpp:25]: Haha, nice catch! it totally would. I moved this function from the bottom of the file but missed this.

**Review by shan-as — APPROVED** [review]

**Review by dvangeyn-as — APPROVED** [review]

**Pull request merged**